File: BOJ/[1083] 소트/JFe/1083.py
import sys
N = int(sys.stdin.readline())
A = [int(x) for x in sys.stdin.readline().split()]
S = int(sys.stdin.readline())
# S번 교환하면서 가장 내림차순이 되게 만들기
for i in range(N):
    max_idx, max_v = i, A[i]
    # 현재 위치에서 최대로 교환 가능한 위치까지 최댓값 구하기
    for j in range(i, i+S+1):
        if j == N: break
        # 최댓값, 최댓값 인덱스 저장
        if max_v < A[j]:
            max_v = A[j]
            max_idx = j
    # 최댓값 위치까지 교환
    for j in range(max_idx, i, -1):
        A[j], A[j-1] = A[j-1], A[j]
        S -= 1
    if S == 0: break    # 더 이상 교환 못하면 break
for i in A: print(i, end=' ')

File: BOJ/[1083] 소트/SUbbb/1083.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    static int[] numbers;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        numbers = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        int S = Integer.parseInt(br.readLine());

        for (int i = 0; i < N; i++) {
            if (S == 0) break;

            int maxIdx = findMax(i, i + S);

            for (int j = maxIdx; j > i; j--) {
                swap(j - 1, j);
                S--;
            }
        }

        for (int n : numbers)
            System.out.print(n + " ");
    }

    private static int findMax(int s, int e) {
        int maxIdx = 0;
        int max = 0;

        for (int i = s; i <= e && i < numbers.length; i++) {
            if (max < numbers[i]) {
                maxIdx = i;
                max = numbers[i];
            }
        }

        return maxIdx;
    }

    private static void swap(int i1, int i2) {
        int tmp = numbers[i1];
        numbers[i1] = numbers[i2];
        numbers[i2] = tmp;
    }
}